exporters/prometheus/exporter.go

```diff
@@ -26,6 +26,7 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
+	"github.com/puzpuzpuz/xsync"
```

Comment:
I would rather not introduce a dependency just to deal with concurrency of a type.

Reply:
Doubly so when [`sync.Map` exists](https://pkg.go.dev/sync#Map).

Reply:
sync.Map isn't a type safe thing unlike this implementation based on generics. However, I agree that we should avoid new dependencies here. Do you want me to:
* use sync.Map
* protect the existing map with sync.RWMutex
* implement a generic map type in this library (potentially could be re-used in other places in this repo)

Reply:
Given the type looks to be passed I would expect a new concurrent map type is needed. That said, I don't know if it needs to be generic given it is for a single use.